### PR TITLE
Add psql and URI commands in project settings

### LINKED ISSUE #4910: Add psql command to connection strings section in database settings
We show Postgres URI as "PSQL connection string".
It is possible to connect from psql using URI, but:

1. This psql connection option is not documented so many users may not know about it.
2. It makes a password visible when you paste it.
3. It does not work for some passwords (I have just accidentally discovered that).

I would like to have a copiable psql command like the following:

`psql -h db.[PROJECT_REF].supabase.co -p 5432 -d postgres -U postgres`

So we may consider adding separate sections with PSQL command and URI like the following:
```
psql:
    psql -h db.{PROJECT_REF}.supabase.co -p 5432 -d postgres -U postgres
URI:
    postgresql://postgres:[YOUR-PASSWORD]@db.[PROJECT_REF].supabase.co:5432/postgres
Golang:
    ...
JDBC:
    ...
...
```

Database settings page:
https://app.supabase.io/project/[PROJECT_REF]/settings/database

### PR DESCRIPTION
**Pull request by imskr**
## What kind of change does this PR introduce?

feature

- Adds `psql` and `uri` connection strings in project settings

Fixes #4910 

## What is the current behavior?
Screenshot_20220126_185722

## What is the new behavior?
Screenshot_20220126_185259
Screenshot_20220126_185243

/cc @kiwicopple @Ziinc

### CODE REVIEW

File: studio/pages/project/[ref]/settings/database.tsx [hunk 1]
```diff
@@ -303,14 +303,18 @@ const GeneralSettings: FC<any> = ({ projectRef }) => {
   const DB_FIELDS = ['db_host', 'db_name', 'db_port', 'db_user', 'inserted_at']
   const connectionInfo = pluckObjectFields(formModel, DB_FIELDS)
 
-  const defaultConnString =
+  const defaultURIConnString =
```
Comment (kiwicopple) [inline studio/pages/project/[ref]/settings/database.tsx:306]: any reason for the change here?

File: studio/pages/project/[ref]/settings/database.tsx [hunk 2]
```diff
@@ -303,14 +303,18 @@ const GeneralSettings: FC<any> = ({ projectRef }) => {
...
     `postgresql://${connectionInfo.db_user}:[YOUR-PASSWORD]@` +
     `${connectionInfo.db_host}:${connectionInfo.db_port.toString()}` +
     `/${connectionInfo.db_name}`
   const golangConnString =
     `user=${connectionInfo.db_user} password=[YOUR-PASSWORD] ` +
     `host=${connectionInfo.db_host} port=${connectionInfo.db_port.toString()}` +
     ` dbname=${connectionInfo.db_name}`
+  const defaultPSQLConnString =
```
Comment (kiwicopple) [inline studio/pages/project/[ref]/settings/database.tsx:314]: should we camelCase this one? `defaultPsqlConnString`

File: studio/pages/project/[ref]/settings/database.tsx (hunk 1 above)
Comment (kiwicopple) [inline studio/pages/project/[ref]/settings/database.tsx:306]: should we camelCase this one? `defaultUriConnString`

File: studio/pages/project/[ref]/settings/database.tsx [hunk 3]
```diff
@@ -303,14 +303,18 @@ const GeneralSettings: FC<any> = ({ projectRef }) => {
...
     `${connectionInfo.db_host}:${connectionInfo.db_port.toString()}` +
     `/${connectionInfo.db_name}`
   const golangConnString =
     `user=${connectionInfo.db_user} password=[YOUR-PASSWORD] ` +
     `host=${connectionInfo.db_host} port=${connectionInfo.db_port.toString()}` +
     ` dbname=${connectionInfo.db_name}`
+  const defaultPSQLConnString = 
+    `psql -h db.` + `${connectionInfo.db_host}` + `.supabase.co -p ` +
+    `${connectionInfo.db_port.toString()}` + ` -d ${connectionInfo.db_name}` +
+    ` -U ${connectionInfo.db_user}`
```
Comment (kiwicopple) [inline studio/pages/project/[ref]/settings/database.tsx:317]: should we make this one 

```
`psql -h db.${connectionInfo.db_host}.supabase.co -p ` +
`${connectionInfo.db_port.toString()} -d ${connectionInfo.db_name} ` +
`-U ${connectionInfo.db_user}`
```

A bit more readable!

**Review by kiwicopple — APPROVED** [review]
Nice one @imskr - thanks for this! I'll leave @Ziinc to add any other suggestions

File: studio/pages/project/[ref]/settings/database.tsx (hunk 2 above)
Comment (imskr) [reply studio/pages/project/[ref]/settings/database.tsx:314]: yes sure

File: studio/pages/project/[ref]/settings/database.tsx (hunk 1 above)
Comment (imskr) [reply studio/pages/project/[ref]/settings/database.tsx:306]: yes

File: studio/pages/project/[ref]/settings/database.tsx (hunk 3 above)
Comment (imskr) [reply studio/pages/project/[ref]/settings/database.tsx:317]: yes this one is more readable, thanks

**Comment by imskr** [comment]
Thanks for the suggestions @kiwicopple I have added the changes.

File: studio/pages/project/[ref]/settings/database.tsx [hunk 4]
```diff
@@ -303,14 +303,18 @@ const GeneralSettings: FC<any> = ({ projectRef }) => {
   const DB_FIELDS = ['db_host', 'db_name', 'db_port', 'db_user', 'inserted_at']
   const connectionInfo = pluckObjectFields(formModel, DB_FIELDS)
 
-  const defaultConnString =
+  const defaultUriConnString =
```
Comment (Ziinc) [inline studio/pages/project/[ref]/settings/database.tsx:306]: let's remove the `default`, does not make sense in this context anymore.

File: studio/pages/project/[ref]/settings/database.tsx [hunk 5]
```diff
@@ -303,14 +303,18 @@ const GeneralSettings: FC<any> = ({ projectRef }) => {
...
     `postgresql://${connectionInfo.db_user}:[YOUR-PASSWORD]@` +
     `${connectionInfo.db_host}:${connectionInfo.db_port.toString()}` +
     `/${connectionInfo.db_name}`
   const golangConnString =
     `user=${connectionInfo.db_user} password=[YOUR-PASSWORD] ` +
     `host=${connectionInfo.db_host} port=${connectionInfo.db_port.toString()}` +
     ` dbname=${connectionInfo.db_name}`
+  const defaultPsqlConnString =
```
Comment (Ziinc) [inline studio/pages/project/[ref]/settings/database.tsx:314]: same as above, lets remove `default` and just keep it to `psqlConnString`.

File: studio/pages/project/[ref]/settings/database.tsx [hunk 6]
```diff
@@ -303,14 +303,18 @@ const GeneralSettings: FC<any> = ({ projectRef }) => {
...
+  const defaultUriConnString =
     `postgresql://${connectionInfo.db_user}:[YOUR-PASSWORD]@` +
     `${connectionInfo.db_host}:${connectionInfo.db_port.toString()}` +
     `/${connectionInfo.db_name}`
   const golangConnString =
     `user=${connectionInfo.db_user} password=[YOUR-PASSWORD] ` +
     `host=${connectionInfo.db_host} port=${connectionInfo.db_port.toString()}` +
     ` dbname=${connectionInfo.db_name}`
+  const defaultPsqlConnString = 
+    `psql -h db.${connectionInfo.db_host}.supabase.co -p ` +
```
Comment (Ziinc) [inline studio/pages/project/[ref]/settings/database.tsx:315]: `.supabase.co` should not be present. `db.` as well.

Should interpolate the `connectionInfo.db_host` directly, see the uri version

**Review by Ziinc — CHANGES_REQUESTED** [review]
Just a minor variable name adjustment. uri interpolation looks off.

File: studio/pages/project/[ref]/settings/database.tsx (hunk 4 above)
Comment (imskr) [reply studio/pages/project/[ref]/settings/database.tsx:306]: okay

**Comment by imskr** [comment]
Thanks @Ziinc I have made the changes :)

**Review by Ziinc — APPROVED** [review]
🙏 looks good

**Pull request merged**